src/appengine/libs/auth.py

```diff
@@ -137,11 +137,17 @@ def get_current_user():
   if iap_email:
     return User(iap_email)
 
-  oauth_email = getattr(current_request, '_oauth_email', None)
+  if flask.request:
+    oauth_email = getattr(flask.g, '_oauth_email', None)
```

Comment:
rather than all these ifs, you can just do it once

if flask.request:
  cache_backing = flask.request
else:
  cache_backing = get_current_request()

and use cache_backing everywhere.